measurement/validation.py

```diff
@@ -242,14 +255,18 @@ def generate_validation_report(
     var = Variable.objects.get(variable_code=variable)
 
     data = get_data_to_validate(
-        station, variable, start_time, end_time, include_validated
+        station, variable, start_time, end_time, include_validated, only_validated
     )
-    suspicious = flag_suspicious_data(data, maximum, minimum, period, var.diff_error)
-    summary = generate_daily_summary(
-        data, suspicious, period, var.null_limit, var.is_cumulative
-    )
-    granular = pd.concat([data, suspicious], axis=1)
-    return summary, granular
+    if not data.empty:
+        suspicious = flag_suspicious_data(
+            data, maximum, minimum, period, var.diff_error
+        )
+        summary = generate_daily_summary(
+            data, suspicious, period, var.null_limit, var.is_cumulative
+        )
+        granular = pd.concat([data, suspicious], axis=1)
+        return summary, granular
+    return pd.DataFrame(), pd.DataFrame()
```

Comment:
I think it is nicer to change the order of things here and keep most of the body of the function unindented:

```suggestion
    if data.empty:
        return pd.DataFrame(), pd.DataFrame()

    suspicious = flag_suspicious_data(
        data, maximum, minimum, period, var.diff_error
    )
    summary = generate_daily_summary(
        data, suspicious, period, var.null_limit, var.is_cumulative
        )
    granular = pd.concat([data, suspicious], axis=1)
    return summary, granular
    
```